### PR TITLE
feat(examples): add `usb-serial` example

### PR DESCRIPTION
**Pull request by kaspar030**
# Description

This adds a usb serial example.
It is basically a port of one of Embassy's [examples](https://github.com/embassy-rs/embassy/blob/main/examples/stm32h7/src/bin/usb_serial.rs).

I tested that it works, with Linux as host, on nrf52840dk, rpi-pico and two stm32 (which are not in main yet).

## Change checklist

- [x] I have cleaned up my commit history and squashed fixup commits.
- [x] I have followed the [Coding Conventions](../book/src/coding-conventions.md).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: examples/usb-serial/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+use embassy_usb::{
+    class::cdc_acm::{CdcAcmClass, State},
+    driver::EndpointError,
+};
+
+#[riot_rs::config(usb)]
+fn usb_config() -> embassy_usb::Config<'static> {
+    let mut config = embassy_usb::Config::new(0xc0de, 0xcafe);
```
Comment (chrysn) [inline examples/usb-serial/src/main.rs:21]: There are codes for use with test devices, and while I don't know off my head where to look these up, but they're probably made up.

Good candidates are [1209](https://pid.codes/1209/) / 7D01 (RIOT Test PID) or 1298 / 000[0-f] (generic test PIDs)

File: examples/usb-serial/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    config.product = Some("USB serial example");
+    config.serial_number = Some("12345678");
+    config.max_power = 100;
+    config.max_packet_size_0 = 64;
+
+    // Required for Windows support.
+    config.composite_with_iads = true;
+    config.device_class = 0xEF;
+    config.device_sub_class = 0x02;
+    config.device_protocol = 0x01;
```
Comment (chrysn) [inline examples/usb-serial/src/main.rs:32]: This implies that we're talking AT commands, which is not only incorrect, but it is also an invitation for NetworkManager and similar to start probing the apparent modem, sending garbage before the device is made available to the user. RIOT OS picks 0 (USB_CDC_PROTOCOL_NONE); 0xFF (vendor-specific) might be a sensible choice as well.

**Comment by kaspar030** [comment]
> gives me `8000 bauds`.

I digged a bit, IMO that can just be ignored.

[upstream code](https://github.com/kaspar030/embassy/blob/584940fc7ea02fe184f3ae346ecaa3d00008c95e/embassy-usb/src/class/cdc_acm.rs#L503-L506) says so. Also, the baud rate can only be set from the host.

File: examples/usb-serial/src/main.rs (hunk 1 above)
Comment (kaspar030) [reply examples/usb-serial/src/main.rs:21]: "code cafe" is certainly made up. :) This is copied from upstream. I'll open an issue for default USB ids.

Comment (kaspar030) [reply examples/usb-serial/src/main.rs:21]: https://github.com/future-proof-iot/RIOT-rs/issues/339

File: examples/usb-serial/src/main.rs (hunk 2 above)
Comment (kaspar030) [reply examples/usb-serial/src/main.rs:32]: This is copied from upstream. the subdescriptor is configures with `USB_CDC_PROTOCOL_NONE` iirc: https://github.com/kaspar030/embassy/blob/584940fc7ea02fe184f3ae346ecaa3d00008c95e/embassy-usb/src/class/cdc_acm.rs#L219

**Pull request merged**